agent/proxy.py

```diff
@@ -233,14 +237,26 @@ def remove_redirect(self, host):
             # default domain
             os.rmdir(host_directory)
 
+    def is_nginx_reloading(self) -> bool:
+        shutting_down_processes = int(
+            self.execute(
+                "ps ef | grep 'nginx: worker process is shutting down' | grep -v grep | wc -l"
+            )
+        )
+        if shutting_down_processes > 0:
+            return True
+
     @step("Reload NGINX")
-    def reload_nginx(self):
+    def reload_nginx(self, skip_if_reloading=False):
+        if skip_if_reloading and self.is_nginx_reloading():
+            return
```

Comment:
If you don't reload here, you still somehow need to reload one more time at the end of everything right?

Alternate:
- for bulk config regen - keep regenerating configs many times
- only issue one reload at end when done regenerating. Doesn't matter MUCH for suspend/archive anyway afaik.

@adityahase

Reply:
> Doesn't matter MUCH for suspend/archive anyway

Idea was this only for skipping the reload this way. Worst case some suspended/archived will get 404 till the next reload